packages/backend-common/src/reading/tree/TarArchiveResponse.ts

```diff
@@ -36,11 +36,13 @@ const pipeline = promisify(pipelineCb);
  */
 export class TarArchiveResponse implements ReadTreeResponse {
   private read = false;
+  public readonly etag;
 
   constructor(
     private readonly stream: Readable,
     private readonly subPath: string,
     private readonly workDir: string,
+    etag: string,
```

Comment:
You can use `private readonly etag: string` as a shorthand like the other. With that you don't need the separate declaration and assignment

Reply:
This etag is `public` so that it can be accessed as a property from outside. I thought `public` things need to be moved at the top unlike `private` ones, but I'll try to group.